logcheck/pkg/logcheck.go

```diff
@@ -614,17 +548,9 @@ func isVerbosityZero(expr ast.Expr) bool {
 	return false
 }
 
-func checkFormatSpecifier(expr *ast.CallExpr, pass *analysis.Pass) bool {
-	if _, ok := expr.Fun.(*ast.SelectorExpr); ok {
-		if _, found := hasFormatSpecifier(expr.Args); found {
-			return true
-		}
-	}
-	return false
-}
-
-// keysCheck check if all keys in keyAndValues are valid keys according to the guidelines.
-func keysCheck(keyValues []ast.Expr, fun ast.Expr, pass *analysis.Pass, funName string) {
+// kvCheck check if all keys in keyAndValues are valid keys according to the guidelines
+// and that the values can be formatted.
+func kvCheck(keyValues []ast.Expr, fun ast.Expr, pass *analysis.Pass, funName string, keyCheckEnabled, parametersCheckEnabled bool) {
```

Comment:
I think this function has grown too big and complicated. What do you think about {
```
func kvCheck(keyValues []ast.Expr, fun ast.Expr, pass *analysis.Pass, funName string, keyCheckEnabled, parametersCheckEnabled bool) {
  checkKVParity(keyValues)
  if keyCheckEnabled || parametersCheckEnabled {
    checkKeyInlinedString(keyValues []ast.Expr, fun ast.Expr, pass *analysis.Pass, funName string)
  } 
  if keyCheckEnabled {
    checkKey(keyValues []ast.Expr, fun ast.Expr, pass *analysis.Pass, funName string)
  }
  if parametersCheckEnabled {
    checkParameters(keyValues []ast.Expr, fun ast.Expr, pass *analysis.Pass, funName string)
  }
}
```